src/executor/query/IndexScanExecutor.cpp

```diff
@@ -28,8 +30,16 @@ folly::Future<Status> IndexScanExecutor::indexScan() {
         DataSet dataSet({"dummy"});
         return finish(ResultBuilder().value(Value(std::move(dataSet))).finish());
     }
+
+    const auto &ictxs = lookup->queryContext();
+    auto iter = std::find_if(
+        ictxs.begin(), ictxs.end(), [](auto &ictx) { return !ictx.index_id_ref().is_set(); });
+    if (ictxs.empty() || iter != ictxs.end()) {
```

Comment:
Is the checking logic in the if statement` ictxs.empty()` necessary?

Reply:
My test cases is for this scenario.